doc/specs/#4999 - Improved keyboard handling in Conpty.md

```diff
@@ -0,0 +1,514 @@
+
+* Conpty[2] will ask for `win32-input-mode` from conpty[1] when conpty[2] first
+  boots up.
+    - As conpty[1] is just a conhost that knows how to handle
+      `win32-input-mode`, it will switch its own VT input handling into
+      `win32-input-mode`
+* When the user types keys in Windows Terminal, WT will translate them into
+  win32 sequences and send them to conpty[1]
+* Conpty[1] will translate those win32 sequences into `INPUT_RECORD`s. When
+  conpty[1] writes these to its buffer, it will translate the `INPUT_RECORD`s
+  into VT sequences for the `win32-input-mode`. This is because it believes the
+  client (in this case, the conpty[2] running attached to `wsl`) wants
+  `win32-input-mode`.
+* When WSL reads the input, it'll read (using `ReadConsoleInput`) a stream of
+  `INPUT_RECORD`s that contain only character information, which it will then
+  use to pass a stream of characters to conpty[2].
+* Conpty[2] will get those sequences, and will translate those win32 sequences
+  into `INPUT_RECORD`s
+* When `cmd.exe` reads the input, they'll receive the full `INPUT_RECORD`s
+  they're expecting
+
+
+## UI/UX Design
+
+This is not a user-facing feature.
+
+## Capabilities
+
+### Accessibility
+
+_(no change expected)_
+
+### Security
+
+_(no change expected)_
+
+### Reliability
+
+_(no change expected)_
+
+### Compatibility
+
+This isn't expected to break any existing scenarios. The information that we're
+passing to conpty from the Terminal should strictly have _more_ information in
+them than they used to. Conhost was already capable of translating
+`INPUT_RECORD`s back into VT sequences, so this should work the same as before.
+
+There's some hypothetical future where the Terminal isn't connected to conpty.
+In that future, the Terminal will still be able to work correctly, even with
+this ConPTY change. The Terminal will only switch into sending
+`win32-input-mode` sequences when _conpty asks for them_. Otherwise, the
+Terminal will still behave like a normal terminal emulator.
+
+### Performance, Power, and Efficiency
+
+_(no change expected)_
+
+## Potential Issues
+
+_(no change expected)_
+
+## Future considerations
+
+* We could also hypothetically use this same mechanism to send Win32-like mouse
+  events to conpty, since similar to VT keyboard events, VT mouse events don't
+  have the same fidelity that Win32 mouse events do.
+  - We could enable this with a different terminating character, to identify
+    which type of `INPUT_RECORD` event we're encoding.
+* Client applications that want to be able to read full Win32 keyboard input
+  from `conhost` _using VT_ will also be able to use `^[[?9001h` to do this. If
+  they emit `^[[?9001h`, then conhost will switch itself into
+  `win32-input-mode`, and the client will read `win32-input-mode` encoded
+  sequences as input. This could enable other cross-platform applications to
+  also use win32-like input in the future.
+
+## Options Considered
+

```

Comment:
Options Considered: use an intermediate char - why not?

Reply:
SGR Mouse uses this.

Reply:
I suppose we could? I didn't really consider this too strongly one way or another. Since we're not terribly worried about the `_` conflict with the SCO sequences, I figure we might as well not waste another `char` on it, right?